### PR TITLE
added etiquette doc

### PR DESCRIPTION
**Pull request by mandric**
Not sure if this is useful or just me venting.

### CODE REVIEW

File: md/dev/mediquette.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    The commit title is a short accurate summary about what code was changed.
+    Someone should be able to read the title of the commit message and have an
+    idea of what part of the code has changed and the impact it might have on
+    things. Use words like: "bugfix", "refactored", "updated" and "fixed" .
+    Use the past tense to signify that the change is done and not something
+    still happening currently or in the future.
+
+    Bad:
+      Fix Typo
```
Comment (SCdF) [inline md/dev/mediquette.txt]: commit titles are now in many cases irrelevant (I have stopped caring about a large number of them) because we are trailing forced squashes now. So my first commit will have a good title and comment, and any fixes after it will be hot garbage (they might as well just be `dsafgafdgfa`) because I know I'm going to squash later.

File: md/dev/mediquette.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     explains appropriate usage.
+
+  2) Never delete or edit someone else's content.  If you think something is
+     inappropriate then notify the person first and ask them politely to remove
+     it.
+
+  3) In general deleting something you already posted is poor etiquette.
+     Because it changes the context of the discussion.  If you have a problem
+     with something you wrote then just create a thread on it or just repost it
+     with your edits.
```
Comment (SCdF) [inline md/dev/mediquette.txt]: I have no issue editing posts to fix typos. I think it adds more clarity and less noise compared to writing additional comments with `*blah` or `s/blag/blah/` style comments

File: md/dev/mediquette.txt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    Use the past tense to signify that the change is done and not something
+    still happening currently or in the future.
+
+    Bad:
+      Fix Typo
+
+    Good: 
+      Fixed typo in xform upload script.
+
+ Commit Bodies
```
Comment (SCdF) [inline md/dev/mediquette.txt]: We've had contrasting conversations about this one, with no authoritative resolution.

What you have written roughly aligned with my beliefs / feelings. However, others believe GH tickets are the single source of truth, and others again think actual raw commited code is.

I don't know if we want to re-hash it, because it feels like a tabs vs. spaces level discussion, that will go nowhere slowly. When we last discussed it I personally resolved to do what I thought was best, and hopefully if my methods are more valid that will become apparent over time (and vice versa).

File: md/dev/mediquette.txt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+     Linking the GitHub issue is also good here but it does not replace the
+     commit body which is the final result from various discussion in various
+     channels (Slack/GitHub/Hangouts).  There is no guarantee that the issue
+     will exist in the future, explain the issue clearly, or even be readable.
+     The commit is closest to the code (written by the programmer) it cannot
+     be replaced by a link to the issue.  Feel free to copy relevant parts
+     from the issue into the commit body.
+
+ Branches
```
Comment (SCdF) [inline md/dev/mediquette.txt]: I'm not sure if this is documented anywhere else, but to be clear we have definitely all decided on feature branches where relevant.

File: md/dev/mediquette.txt [hunk 5]
```diff
@@ -0,0 +1,74 @@
+
+Etiquette @medic
```
Comment (estellecomment) [inline md/dev/mediquette.txt]: Interesting! 

I would add something along the lines of "These are guidelines, and not set in stone. Use your own judgement." (it allows some flexibility)

File: md/dev/mediquette.txt [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+Be kind to the plants and animals.
```
Comment (estellecomment) [inline md/dev/mediquette.txt]: "NB: humans are animals too. "
:)

Comment (estellecomment) [inline md/dev/mediquette.txt]: I would add "Try your hardest not to be an asshole." That to me is a requirement for working with people. That cuts short to any aggressive or passive-aggressive behavior.

File: md/dev/mediquette.txt [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Slack
+
+  1) Try to pay attention what channel you are writing in and keep noise
+     (tangential or unrelated discussion) to a minimum.  Use threads when
+     appropriate to keep tangential discussion contained and out of the main
+     channel.   Each channel should have a purpose with a description that
+     explains appropriate usage.
+
+  2) Never delete or edit someone else's content.  If you think something is
```
Comment (estellecomment) [inline md/dev/mediquette.txt]: +1
I think only only admins have this ability as well, so it's kinda awkward for that too.

File: md/dev/mediquette.txt (hunk 2 above)
Comment (estellecomment) [inline md/dev/mediquette.txt]: Hmmmm. I'd say it's impractical to delete your own messages, but not poor etiquette. As in, it doesn't actually cause harm. 
Is this related to a particular use case?

File: md/dev/mediquette.txt (hunk 5 above)
Comment (estellecomment) [inline md/dev/mediquette.txt]: I find this doc very interesting. 
I'd say that slack etiquette vs. git commit formats are two different deals, and would probably split out the part about git commit into the workflow doc for instance.

Comment (estellecomment) [inline md/dev/mediquette.txt]: Would the etiquette doc be a could place for conflict resolution guidelines @mandric ? Or do you see it as separate?

File: md/dev/mediquette.txt [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+     and related issues that might be affected.  
+
+     The product team benefits from this input by getting a preface of the
+     technical issues without having to read code.  The code review is easier
+     because some of the logic that might be hard  to distill from a code read
+     is stated in the commit body. 
+
+     Linking the GitHub issue is also good here but it does not replace the
+     commit body which is the final result from various discussion in various
+     channels (Slack/GitHub/Hangouts).  There is no guarantee that the issue
```
Comment (estellecomment) [inline md/dev/mediquette.txt]: > There is no guarantee that the issue will exist in the future

Well. I guess there could be a situation where github dies and git stays alive. Personally I use github heavily (including for things I could be doing with git), and don't really plan for github issues disappearing.

File: md/dev/mediquette.txt (hunk 3 above)
Comment (estellecomment) [inline md/dev/mediquette.txt]: > it feels like a tabs vs. spaces level discussion

+1
It's down to whether you like history recorded in issues or commits. 

I don't have strong thoughts, so i guess I could start writing commit bodies if necessary...

File: md/dev/mediquette.txt (hunk 1 above)
Comment (mandric) [inline md/dev/mediquette.txt]: Yeah that's a good point!  This really only applies to what ends up in `master` branch, not your messy working branch which you can do anything you want in.  So this does apply to the squash commit then.

File: md/dev/mediquette.txt (hunk 6 above)
Comment (estellecomment) [inline md/dev/mediquette.txt]: "Being grumpy on bad days is accepted."

File: md/dev/mediquette.txt (hunk 4 above)
Comment (mandric) [inline md/dev/mediquette.txt]: Here's what we have documented re branches: https://github.com/medic/medic-docs/blob/master/md/dev/workflow.md#commits it probably makes sense to move the git related stuff from here into the workflow doc.

**Comment by mandric** [comment]
FYI feel free to edit (collaborate) on this branch if you're so inspired.

File: md/dev/mediquette.txt (hunk 5 above)
Comment (mandric) [inline md/dev/mediquette.txt]: Conflict resolution guidelines?  I'm not sure what that entails, but maybe write something about and then we'd get an idea of where it belongs?

File: md/dev/mediquette.txt (hunk 6 above)
Comment (mandric) [inline md/dev/mediquette.txt]: Hahaha, I like that!

File: md/dev/mediquette.txt (hunk 2 above)
Comment (mandric) [inline md/dev/mediquette.txt]: It changes context which bothers me because then it potentially changes the meaning of someone's response... can potentially change the entire conversation, it just feels like dangerous territory, slippery slope that should be avoided or reserved for specific incidents.  I feel like editing and deleting posts needs some treatment here and shouldn't be left completely arbitrary.

File: md/dev/mediquette.txt (hunk 1 above)
Comment (mandric) [inline md/dev/mediquette.txt]: Absolutely, I should add a note that when in your working branch these guidelines are more flexible because you still have one more edit phase during squash/merge before it goes to master.  So then it's back to use your judgement.

File: md/dev/mediquette.txt (hunk 8 above)
Comment (mandric) [inline md/dev/mediquette.txt]: For me Git is completely independent of GitHub and at any point we should be able to move services if necessary.  I see GitHub as temporary storage and Git as permanent storage.  At any point we should be able to move to another project/issue mgmt tool without affecting our code/knowledge base/docs/etc.

File: md/dev/mediquette.txt (hunk 3 above)
Comment (garethbowen) [inline md/dev/mediquette.txt]: Agree I don't want to get into it here, but I for one prefer documentation in GH issues. I really like that I can add formatting, links, images, etc. I guess I feel if something needs to be in a git commit body it should just be a comment in the code (or more readable code if possible!).

File: md/dev/mediquette.txt (hunk 2 above)
Comment (garethbowen) [inline md/dev/mediquette.txt]: IMO it depends. I delete stuff from time to time, but usually when nobody else has responded yet to avoid changing the context. Mostly I delete spurious failed build messages (eg: when the branch has been deleted).

**Comment by garethbowen** [comment]
It'd be good to add something about meeting etiquette as well, eg: try your hardest to arrive on time with working mic and audio.

File: md/dev/mediquette.txt (hunk 2 above)
Comment (browndav) [inline md/dev/mediquette.txt]: I don't want to prohibit deletions or edits of peoples' own messages. There are perfectly valid reasons to do it  – e.g. you've posted a line of questions that turns out to be easily-answered, haven't seen any responses, and later discover it was obvious and would waste the time of others. Same thing for inaccurate information: you might want to edit it to ensure someone doesn't find it later and do the wrong thing. I'd probably outline what you consider good judgement rather than prohibiting use of a Slack feature.

File: md/dev/mediquette.txt (hunk 8 above)
Comment (browndav) [inline md/dev/mediquette.txt]: GitHub isn't temporary storage for us – it contains substantial and useful history in the form of issues, is something we back up, and is something that we would never wipe out the history for. We have no plans to move services or ever treat issues as ephemeral things.

File: md/dev/mediquette.txt (hunk 6 above)
Comment (browndav) [inline md/dev/mediquette.txt]: There is a substantial difference between being grumpy and being grumpy *at other people* who are doing their best to help. The former is fine. The latter must be prohibited – it's contagious and destructive, and IMO is better solved by just taking some time away and doing what you need to do to return to normal.

File: md/dev/mediquette.txt (hunk 5 above)
Comment (browndav) [inline md/dev/mediquette.txt]: Top-level request: http://archive.fortune.com/galleries/2008/fortune/0804/gallery.bestadvice.fortune/7.html. I think the soda company CEO was beaten to this advice by a couple millennia, but IMO it's important.

File: md/dev/mediquette.txt (hunk 6 above)
Comment (mandric) [inline md/dev/mediquette.txt]: I prefer "discouraged" over "prohibited", not sure why that language bothers me exactly I think we can't really prohibit anything unless it's against the law?  

I think being grumpy at other people is the definition of grumpy, otherwise you are just upset and nobody is around.  Once people are around you then you project your mental state on them and become  Señor Grumpypants.  Which amongst friends is ok, but it's not the more productive role to be in a professional environment or to get stuff done, but people deal with it like any barrier to success.  We should encourage people to play the role of Big Bird, not Oscar, when you encounter a grumpy peer.

File: md/dev/mediquette.txt (hunk 8 above)
Comment (mandric) [inline md/dev/mediquette.txt]: @browndav do you think including more information in the git history (e.g. commit bodies with complete information) is useful?  Any other opinions about Git etiquette?

**Comment by mandric** [comment]
We had to move this to a private google doc.
